contracts/utils/fileSystem.js

```diff
@@ -0,0 +1,31 @@
+const fs = require("fs");
+const papa = require("papaparse");
+const { ethers } = require("hardhat");
+
+
+const parseCsv = (filePath) => {
+  const csvFile = fs.readFileSync(filePath);
+  const csvData = csvFile.toString();
+
+  return new Promise((resolve) => {
+    papa.parse(csvData, {
+      header: true,
+      complete: ({ data }) => {
+        console.log("Complete", data.length, "records.");
+        resolve(data);
+      },
+    });
+  });
+};
+
+const hashFileContents = async (filePath) => {
```

Comment:
Why is this async? ethers.utils.keccak256 is not async.

Reply:
thanks changing